db/lock.go

```diff
@@ -58,21 +57,23 @@ func setDocumentLocked(id string, upsert bool) (bool, error) {
 	// for findAndModify-ing the lock
 
 	// timeout to check for
-	timeoutThreshold := time.Now().Add(-LockTimeout)
+	timeoutThreshold := time.Now().Add(-lockTimeout)
 
 	// construct the selector for the following cases:
 	// 1. lock is not held by anyone
 	// 2. lock is held but has timed out
 	selector := bson.M{
-		"_id": GlobalLockId,
-		"$or": []bson.M{{"locked": false}, {"locked_at": bson.M{"$lte": timeoutThreshold}}},
+		"_id": id,
+		"$or": []bson.M{
+			{"locked": false},
+			{"locked_at": bson.M{"$lte": timeoutThreshold}},
+		},
 	}
 
 	// change to apply to document
 	change := mgo.Change{
 		Update: bson.M{"$set": bson.M{
 			"locked":    true,
-			"locked_by": id,
 			"locked_at": time.Now(),
```

Comment:
Are there any risks of using the system clock like this?

Would using doing all the time handling stuff on MongoDB server be safer?

Reply:
Of course. 

The theory is that 8 minutes is enough of a window for timeout to protect against glitchy clockskew. 

Having said that, I think that this lock implementation has never been particularly watertight, which is part of the reason why this code is legacy and we're moving away from this pararallelism model. 

Regardless, this is outside of the scope of this code review.